File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int size = nums.size();
        int count = 1;

        int left = 0, right = 0;
        for(int i=0; i<size-1; i++)
        {
            if(nums[i] != nums[i+1])
            {
                nums[left+1] = nums[i+1];
                right = i+1;
                left = left+1;
                count++;
            }
            else if(nums[i] == nums[+1])
            {
                right = i+1;
            }
        }
        return count;
    }
};

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int size = nums.length;

        // EDGECASE: IF THE SIZE THE ARRAY IS 1
        if (size == 1) {
            return 1;
        }

        // NON-EDGECASE
        int start = 0;
        int end = start+1;

        while (end < size) {
            if (nums[start] == nums[end]) {
                end++;
            } else {
                nums[start+1] = nums[end];
                start++;
                end++;
            }
        }

        return start+1;
    }
}